packages/common/src/api/hooks/usePaginatedQuery.ts

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from 'react'
+
+import { isEqual } from 'lodash'
+import { useCustomCompareEffect } from 'react-use'
+
+import { Status } from 'models/Status'
+
+import { QueryHookResults } from '../types'
+
+export const usePaginatedQuery = <
+  Data extends [],
+  ArgsType extends { limit: number; offset: number },
+  HookType extends (args: ArgsType) => QueryHookResults<Data>
+>(
+  useQueryHook: HookType,
+  baseArgs: Exclude<ArgsType, 'limit' | 'offset'>,
+  pageSize: number
+) => {
+  const [page, setPage] = useState(0)
+  const args = { ...baseArgs, limit: pageSize, offset: page * pageSize }
+  const result = useQueryHook(args)
+  return {
+    ...result,
+    loadMore: () => setPage(page + 1),
+    hasMore:
+      result.status === Status.IDLE ||
+      (!result.data && result.status === Status.LOADING) ||
+      result.data?.length < pageSize
+  }
+}
+
+export const useAllPaginatedQuery = <
+  Data,
+  ArgsType extends { limit: number; offset: number },
+  HookType extends (args: ArgsType) => QueryHookResults<Data[]>
+>(
+  useQueryHook: HookType,
```

Comment:
Still not entirely sure this doesn't break the rules of hooks. If the caller were to change the hook being passed, then hooks would be getting called in a different order

```
useAllPaginatedQuery(something ? useGetTransactions : useGetTracks, {}, 50)
```

Not sure if it's actually a problem though